### ISSUE TITLE
Update NACL Subnet Association Returns Error - InvalidAssociationID.NotFound

### ISSUE THREAD

**Issue by scottclk** [terraform-providers/terraform-provider-aws#12922]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```
TF: v0.12.9
AWS: v2.58.0
```

### Affected Resource(s)

* aws_network_acl

### Terraform Configuration Files

Create Subnet & Two NACLs with one NACL associated with a single subnet. [Works]
```hcl
resource "aws_network_acl" "test_acl_01" {
  vpc_id     = var.vpc_id
  subnet_ids = [aws_subnet.test_subnet.id]
  tags = {
    Name = "test_acl_01"
  }
}

resource "aws_network_acl" "test_acl_02" {
  vpc_id     = var.vpc_id
  subnet_ids = []
  tags = {
    Name = "test_acl_02"
  }
}

resource "aws_subnet" "test_subnet" {
  vpc_id     = var.vpc_id
  cidr_block = "10.0.0.0/24"

  tags = {
    Name = "test_subnet"
  }
}
```

Move subnet association to other NACL. [Error]
```hcl
resource "aws_network_acl" "test_acl_01" {
  vpc_id     = var.vpc_id
  subnet_ids = []
  tags = {
    Name = "test_acl_01"
  }
}

resource "aws_network_acl" "test_acl_02" {
  vpc_id     = var.vpc_id
  subnet_ids = [aws_subnet.test_subnet.id]
  tags = {
    Name = "test_acl_02"
  }
}

resource "aws_subnet" "test_subnet" {
  vpc_id     = var.vpc_id
  cidr_block = "10.0.0.0/24"

  tags = {
    Name = "test_subnet"
  }
}
```

### Debug Output
Error: InvalidAssociationID.NotFound: The association ID 'aclassoc-xxxxxxxxxxxxxxxxx' does not exist
status code: 400, request id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

### Expected Behavior
After updating the `subnet_ids` to the second NACL the association should be dropped from the first NACL and associated to the second (or vice-versa).

### Actual Behavior
After the error causes the apply to fail the console shows that it has actually associated with aws_network_acl.test_acl_02 and running a second attempt at apply normally resolves everything.
This issue can be sporadic and even work in rare occasions. 

### Steps to Reproduce

1. `terraform apply`
2. Update `subnet_ids` to other NACL.
3. `terraform apply`

**Comment by scottclk** [comment]
I've create a PR that fixes this issue here https://github.com/terraform-providers/terraform-provider-aws/pull/13382

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.63.0 of the Terraform AWS Provider, likely tomorrow. Thanks to @scottclk for the implementation. 👍
